[libxcrypt] build failure

### Operating system

Linux

### Compiler

gcc (GCC) 15.0.0 20241005 (experimental)

### Steps to reproduce the behavior

vcpkg install libxcrypt

### Failure logs

```
./../src/v4.4.36-645fb55159.clean/lib/util-base64.c:24:3: error: initializer-string for array of ‘unsigned char’ is too long [-Werror=unterminated-string-initialization]
   24 |   "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:2116: lib/libcrypt_la-util-base64.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:1349: all] Error 2
```


lib/util-base64.c:

```c
/* Base64-related utility functions and data.  */

#include "crypt-port.h"

const unsigned char ascii64[65] =
  "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
/* 0000000000111111111122222222223333333333444444444455555555556666 */
/* 0123456789012345678901234567890123456789012345678901234567890123 */
  "\x00";
```

```python
>>> len("./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
64
```

### Additional context

_No response_

--- a/ports/libxcrypt/portfile.cmake
+++ b/ports/libxcrypt/portfile.cmake
@@ -1,43 +1,27 @@
 set(VCPKG_POLICY_ALLOW_RESTRICTED_HEADERS enabled)
 
-# Requirements from https://github.com/besser82/libxcrypt?tab=readme-ov-file#build-requirements-and-instructions
-message(
-"${PORT} currently requires the following libraries from the system package manager:
-    autoconf automake libtool pkg-config
-These can be installed on Debian systems via sudo apt install autoconf automake libtool pkg-config"
-)
-find_program(AUTORECONF_BIN autoreconf)
-if(NOT AUTORECONF_BIN)
-    message(FATAL_ERROR "${PORT} requires autoconf from the system package manager (example: \"sudo apt install autoconf\")")
-endif()
-find_program(LIBTOOL_BIN libtoolize)
-if(NOT LIBTOOL_BIN)
-    message(FATAL_ERROR "${PORT} requires libtool from the system package manager (example: \"sudo apt install libtool\")")
-endif()
+message(STATUS "${PORT} requires libltdl-dev from the system package manager (example: \"sudo apt install libltdl-dev\")")
 
-vcpkg_download_distfile(PATCH_FIX_ERROR_STRICT_OVERFLOW
-    URLS https://github.com/besser82/libxcrypt/commit/7fc153170ea6c2938c0392794778de7ec995f8f9.patch?full_index=1
-    SHA512 55f4709c52f6d0a29f159348821c06e3f5df0fae83f487b7a52ce61cd3f6a3a0f48023159fb7029d0a5a3decee36ac6a429cc91e23a83996a3265d681fa11929
-    FILENAME besser82-libxcrypt-7fc153170ea6c2938c0392794778de7ec995f8f9.patch
-)
+vcpkg_find_acquire_program(PERL)
+set(ENV{PERL} "${PERL}")
+
+vcpkg_find_acquire_program(PKGCONFIG)
+set(ENV{PKG_CONFIG} "${PKGCONFIG}")
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO besser82/libxcrypt
     REF "v${VERSION}"
-    SHA512 61e5e393654f37775457474d4170098314879ee79963d423c1c461e80dc5dc74f0c161dd8754f016ce96109167be6c580ad23994fa1d2c38c54b96e602f3aece
-    PATCHES
-        "${PATCH_FIX_ERROR_STRICT_OVERFLOW}"
+    SHA512 05b0288ca1f1371674516df2e6cc9034f34057bb86a4b1702577dcf1eb7ce4730fad3e660d69123f108aec1f1ab8a0f84aec50ada012fe523e94d10e2303835e
 )
 
-vcpkg_configure_make(
+vcpkg_make_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    AUTOCONFIG
+    AUTORECONF
 )
-
-vcpkg_install_make()
+vcpkg_make_install()
 vcpkg_fixup_pkgconfig()
 
-vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSING" "${SOURCE_PATH}/COPYING.LIB")
-
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSING" "${SOURCE_PATH}/COPYING.LIB")

--- a/ports/libxcrypt/vcpkg.json
+++ b/ports/libxcrypt/vcpkg.json
@@ -1,9 +1,14 @@
 {
   "name": "libxcrypt",
-  "version": "4.4.36",
-  "port-version": 2,
+  "version": "4.4.38",
   "description": "libxcrypt is a modern library for one-way hashing of passwords. On Linux-based systems, by default libxcrypt will be binary backward compatible with the libcrypt.so.1 shipped as part of the GNU C Library.",
   "homepage": "https://github.com/besser82/libxcrypt",
-  "license": "LGPL-2.1-only",
-  "supports": "linux"
+  "license": null,
+  "supports": "linux",
+  "dependencies": [
+    {
+      "name": "vcpkg-make",
+      "host": true
+    }
+  ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5437,8 +5437,8 @@
       "port-version": 0
     },
     "libxcrypt": {
-      "baseline": "4.4.36",
-      "port-version": 2
+      "baseline": "4.4.38",
+      "port-version": 0
     },
     "libxcvt": {
       "baseline": "0.1.2",

--- a/versions/l-/libxcrypt.json
+++ b/versions/l-/libxcrypt.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "823052bfe81904cdf887fc83bbbf57eec872d057",
+      "version": "4.4.38",
+      "port-version": 0
+    },
+    {
       "git-tree": "825ff0426ce377f1e386f7effd1f3dfb5e218a64",
       "version": "4.4.36",
       "port-version": 2

